### DISCUSSION TITLE
Version 1.0.0rc1

### DISCUSSION THREAD

**Discussion by aryan26roy** [scikit-hep/formulate#49] [General]
## Summary
This is the first release after the switch to Lark.

## What's Changed
* ci: add trusted publisher and attestations by @henryiii in https://github.com/scikit-hep/formulate/pull/44
* Fix trusted publisher delivery by @jonas-eschle in https://github.com/scikit-hep/formulate/pull/47
* Update version and .gitignore by @aryan26roy in https://github.com/scikit-hep/formulate/pull/48

## New Contributors
* @aryan26roy made their first contribution in https://github.com/scikit-hep/formulate/pull/48

**Full Changelog**: https://github.com/scikit-hep/formulate/compare/v0.1.1...v1.0.0rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/scikit-hep/formulate/releases/tag/v1.0.0rc1'>Version 1.0.0rc1</a>.</em>

**Comment by jonas-eschle** [comment]
I was just wandering @aryan26roy what's the plan for the 1.0 release? How long do you wanna keep the RC only?
Also, I would suggest to add it to conda-forge once 1.0 is out, I can do that and add you as a maintainer as well, okay?

**Reply by aryan26roy** [reply]
Hey @jonas-eschle , for now I only have one action item to complete before v1.0 release, that is updating the standalone scripts with the latest version of Lark. I was hoping to wait a bit and see if anyone has any other suggestions before I go for the major version release. Maybe @jpivarski can help us out here and add any other action items he would like to see go in before v1.0.

As for adding it to conda-forge, that is a great idea! I certainly don't mind being added as a maintainer, however, please keep in mind there is only so much time I can dedicate to the maintenance of this project, if it starts amassing users and we start getting a lot of feature-requests/bug-fix requests, I won't be able to keep up.

**Reply by jpivarski** [reply]
> Maybe @jpivarski can help us out here and add any other action items he would like to see go in before v1.0.

I can't think of anything that's missing.

**Reply by jonas-eschle** [reply]
I think the one thing that would break at least my code is the `variable` attribute. Since it's parsing the expressions, it seems reasonable to know about them (similar to how numexpr, sympy or any other expression knows about its input variables).

No strong opinion, but this is the one thing I've been regularly using from formulate (i.e. to let uproot know which variables to load or which variables I will need from a df etc) and seems fairly simpel, as we're parsing the expression anyways (the operations are probably harder/longer)

Other than that it's fine I think

**Reply by aryan26roy** [reply]
Hi @jonas-eschle , I think the functionality to know about variables can be easily added to the current code. It would have to be a recursive call over the whole AST, perhaps something we can do internally before we give the AST object to the end user. @jpivarski I think we can recreate the exact API with this new implementation since they were originally creating a set of all variables and exposing it as a data member, unless I missed something.

**Reply by jonas-eschle** [reply]
Hm, yes, that could make sense. I think it's not a stopping item IMHO, but probably the one most reasonable feature for something like a formula (but the argument can be made that any of the libraries, numexpr or ROOT should be able to parse a corresponding string anyways, so that's somewhat of a doublication, as one of the libraries will always be used (otherwise formulate is meaningless)

**Reply by jonas-eschle** [reply]
@aryan26roy I think enough time has passed and I would suggest to make the 1.0 release?

As mentioned, the only thing could be to add the `variable` attribute, by just using numexpr (as a runtime dependency). It's quick and dirty, but probably the only thing that actually is needed from the old API, even thinking about it since a while now, that is the only thing I would still miss.
